pkg/gadgets/trace/open/tracer/tracer_test.go

```diff
@@ -0,0 +1,216 @@
+		fd, err := unix.Open("/dev/null", 0, 0)
+		if err != nil {
+			return fmt.Errorf("error opening file: %w", err)
+		}
+
+		fileFd = fd
+
+		unix.Close(fd)
+
+		return nil
+	}
+
+	testRunner := utilstest.NewTestRunner[types.Event](t)
+
+	tests := map[string]utilstest.Test[types.Event]{
+		"test without filtering": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				tracer, err := NewTracer(&Config{}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("error creating tracer: %s", err)
+				}
+				return tracer
+			},
+			GenerateEvent: generateEvent,
+			GetAnyExpectedEvent: func(info *utilstest.Info) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					MountNsID: info.MountNsID,
+					Pid:       uint32(info.Pid),
+					UID:       uint32(info.UID),
+					Comm:      info.Comm,
+					Fd:        fileFd,
+					Ret:       fileFd,
+					Err:       0,
+					Path:      "/dev/null",
+				}
+			},
+		},
+		"test filtering (no event should be captured)": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				m, err := utilstest.CreateMntNsFilterMap(0)
+				if err != nil {
+					testRunner.Fatalf("error creating ebpf map: %s", err)
+				}
+
+				tracer, err := NewTracer(&Config{MountnsMap: m}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("error creating tracer: %s", err)
+				}
+				return tracer
+			},
+			GenerateEvent: generateEvent,
+		},
+		"test filtering": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				m, err := utilstest.CreateMntNsFilterMap(info.MountNsID)
+				if err != nil {
+					testRunner.Fatalf("error creating ebpf map: %s", err)
+				}
+
+				tracer, err := NewTracer(&Config{MountnsMap: m}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("error creating tracer: %s", err)
+				}
+				return tracer
+			},
+			GenerateEvent: generateEvent,
+			GetExpectedEvent: func(info *utilstest.Info) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					MountNsID: info.MountNsID,
+					Pid:       uint32(info.Pid),
+					UID:       uint32(info.UID),
+					Comm:      info.Comm,
+					Fd:        fileFd,
+					Ret:       fileFd,
+					Err:       0,
+					Path:      "/dev/null",
+				}
+			},
+		},
+		"test UID": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				m, err := utilstest.CreateMntNsFilterMap(info.MountNsID)
+				if err != nil {
+					testRunner.Fatalf("error creating ebpf map: %s", err)
+				}
+
+				tracer, err := NewTracer(&Config{MountnsMap: m}, nil, eventCallback)
+				if err != nil {
+					testRunner.Fatalf("error creating tracer: %s", err)
+				}
+				return tracer
+			},
+			RunnerConfig:  &utilstest.Config{UID: 1435},
```

Comment:
Why `1435`?

Reply:
This looks to me like anonymous magic number, so perhaps we could at least put it in a variable like `randomUnprivilegedUID` or something to give it some context.

Reply:
I added a comment where this number is used. If we find later on more instances we could define the constant.